solutions/web3-data-fetching/pages/index.tsx

```diff
@@ -0,0 +1,123 @@
+import Head from 'next/head'
+import { Layout, Text, Page, Link } from '@vercel/examples-ui'
+import { ethers } from 'ethers'
+import useSWR from 'swr'
+import { FC } from 'react'
+import abi from '../lib/BAYC.abi.json'
+import { BORED_APE_YATCH_CLUB_ADDRESS } from '../constants'
+
+const contract = new ethers.Contract(
+  BORED_APE_YATCH_CLUB_ADDRESS,
+  abi,
+  ethers.getDefaultProvider()
+)
+
+const Snippet: FC = ({ children }) => {
+  return (
+    <pre className="border-accents-2 border rounded-md bg-white overflow-x-auto p-6 transition-all">
+      {children}
+    </pre>
+  )
+}
+
+function Home({ contractName }: { contractName: string }) {
+  const { data } = useSWR('name', () => contract.name())
+
+  return (
+    <Page>
+      <Head>
+        <title>Data from smart contracts - Vercel Example</title>
+        <meta
+          name="description"
+          content="Vercel example how to use data from smart contracts"
+        />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+
+      <section className="flex flex-col gap-6">
+        <Text variant="h1">Fetching data from smart contracts</Text>
+        <Text>
+          Smart contracts contain relevant information to applications built on
+          top of blockchains that can run the{' '}
+          <Link href="https://ethereum.org/en/developers/docs/evm/">
```

Comment:
Change this link to open on a new tab, and the other links that go to external sites in this page too.